subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestVerificationFailureHandlingIntegrationTest.groovy

```diff
@@ -43,6 +43,7 @@ class TestVerificationFailureHandlingIntegrationTest extends AbstractIntegration
 
     def 'task does not execute when it has a test task output dependency and VM exits unexpectedly'() {
         given:
+        executer.withStackTraceChecksDisabled()
         withFatalTestExecutionError()
```

Comment:
Hi @blindpirate - I also saw flakiness but only on windows agents; please see the body of `withFatalTestExecutionError()`.

Are you seeing flakiness on non-Windows hosts too?

Reply:
Yes: https://builds.gradle.org/viewLog.html?buildId=47249294

Reply:
I updated the code.

Reply:
In that case, may I suggest this as a better general fix?

```
Index: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestVerificationFailureHandlingIntegrationTest.groovy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestVerificationFailureHandlingIntegrationTest.groovy b/subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestVerificationFailureHandlingIntegrationTest.groovy
--- a/subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestVerificationFailureHandlingIntegrationTest.groovy	(revision 3e7d6b537a3dbfd41a28f9c44f9c668b1f0aef27)
+++ b/subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestVerificationFailureHandlingIntegrationTest.groovy	(date 1637021624407)
@@ -124,9 +124,7 @@
      * Cause the test VM to fail at startup by providing an invalid JVM argument.
      */
     def withFatalTestExecutionError() {
-        if (OperatingSystem.current().windows) {
-            executer.withStackTraceChecksDisabled() // ignore additional ST only seen on Windows
-        }
+        executer.withStackTraceChecksDisabled()
         withPassingTest()
         buildFile << '''
             tasks.named('test', Test).configure {

```

Reply:
Yes, of course, please take a look at the updated code.